### PR TITLE
OCM-4777 | feat: provide common func for password generation and hash

### PR DESCRIPTION
**Pull request by marcolan018**
https://issues.redhat.com/browse/OCM-4777
Signed-off-by: marcolan018 <llan@redhat.com>

### CODE REVIEW

File: pkg/idp/utils/password_utils_test.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package utils
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Password Utils", func() {
```
Comment (den-rgb) [inline pkg/idp/utils/password_utils_test.go:8]: Can we have a test for the failure scenarios too?

**Comment by den-rgb** [comment]
Could something like this be a better implementation of the password generator:
```func GenerateRandomPassword() (string, error) {
	const (
		lowerLetters = "abcdefghijkmnopqrstuvwxyz"
		upperLetters = "ABCDEFGHIJKLMNPQRSTUVWXYZ"
		digits       = "23456789"
		all          = lowerLetters + upperLetters + digits
	)
	password := make([]rune, MaxPasswordLength)
	for i := range password {
		n, err := rand.Int(rand.Reader, big.NewInt(int64(len(all))))
		if err != nil {
			return "", err
		}
		password[i] = rune(all[n.Int64()])
	}

	for _, replace := range []int{5, 11, 17} {
		if replace < len(password) {
			password[replace] = '-'
		}
	}

	return string(password), nil
}```

- The shuffling in the original doesn't really achieve anything other than add to the complexity of the code
- `if password == ""` will always fail after the first iteration so is it really needed?
- The implementation also doesn't guarantee the inclusion of lowerLetters, upperLetters, digits in the password which could be a problem, after running the test a couple of times i noticed some passwords without digits.
- Due to `if i < MaxPasswordLength-1` the last character is always placed at the last position, I've modified it above so that it's shuffled in at random also.

If this was the agreed upon implementation and would rather not have it changed then LGTM

**Comment by marcolan018** [comment]
@den-rgb thanks for the comments
The GenerateRandomPassword() func is extracted from https://github.com/openshift/rosa/blob/master/cmd/create/admin/cmd.go#L176-L209, which is 3 years old.

The purpose of this change is to move the func from the rosa repo to ocm-common, so that it can be used in different clients(ROSA CLI & Terraform Provider). I prefer not to update the inside logic of this func in this PR, to keep the change simple and avoid potential risk. We can work together later to make the refactoring. Is it ok for you?

**Comment by den-rgb** [comment]
@marcolan018 Sounds good :)
lgtm

**Review by ciaranRoche — APPROVED** [review]

**Pull request merged**